common/src/service/HttpClientHelper.cs

```diff
@@ -919,5 +894,39 @@ public void Dispose()
             Task<Exception> exception = mapToExceptionFunc(response);
             return await exception.ConfigureAwait(false);
         }
+
+        internal static HttpClient CreateDefaultClient(IWebProxy webProxy, Uri baseUri, int connectionLeaseTimeoutMilliseconds)
+        {
+#pragma warning disable CA2000 // Dispose objects before losing scope (object is returned by this method, so the caller is responsible for disposing it)
+#if NETCOREAPP && !NETCOREAPP2_0 && !NETCOREAPP1_0 && !NETCOREAPP1_1
+            // SocketsHttpHandler is only available in netcoreapp2.1 and onwards
+            SocketsHttpHandler httpMessageHandler = new SocketsHttpHandler();
+            httpMessageHandler.SslOptions.EnabledSslProtocols = TlsVersions.Instance.Preferred;
+#else
+            HttpClientHandler httpMessageHandler = new HttpClientHandler();
+#endif
+#pragma warning restore CA2000 // Dispose objects before losing scope
+
+#if !NET451 && !NETCOREAPP
+            httpMessageHandler.SslProtocols = TlsVersions.Instance.Preferred;
+            httpMessageHandler.CheckCertificateRevocationList = TlsVersions.Instance.CertificateRevocationCheck;
```

Comment:
nit: I know all these if-def blocks are super messy, but I think I block is missing here:

net451 -> httpclient handler (without ssl protocol and revocation list check)
all other netfm -> httpclient handler (with ssl protocol and revocation list check)
upto netcoreapp 2.0 -> httpclient handler (ssl protocol and revocation list check - available, but missing)
netcoreapp2.1 and above -> sockets http handler (with ssl options)

Reply:
I gave this a rewrite, can you take another look?

Reply:
much better!